vcpkg no longer builds on MSVC 2015

**Describe the bug**
vcpkg no longer builds on MSVC 2015, presumably due to a33044c18637d3df6de93128d33dae1fb5ba575c 

**Environment**
- OS: Windows
- Compiler: MSVC 2015 from AppVeyor

**To Reproduce**
- git clone https://github.com/microsoft/vcpkg
- cd vcpkg
- bootstrap-vcpkg.bat

**Failure logs** 
See https://ci.appveyor.com/project/OSGeo/gdal/builds/30695343/job/souj2w5dyjpldy1d
```
C:\projects\gdal\vcpkg\toolsrc\include\vcpkg/parse.h(114): error C2440: 'return': cannot convert from 'initializer list' to 'vcpkg::Parse::TextRowCol' (compiling source file ..\src\vcpkg\logicexpression.cpp) [C:\projects\gdal\vcpkg\toolsrc\vcpkglib\vcpkglib.vcxproj]
  C:\projects\gdal\vcpkg\toolsrc\include\vcpkg/parse.h(114): note: No constructor could take the source type, or constructor overload resolution was ambiguous (compiling source file ..\src\vcpkg\logicexpression.cpp)
```

--- a/toolsrc/include/vcpkg/textrowcol.h
+++ b/toolsrc/include/vcpkg/textrowcol.h
@@ -4,8 +4,8 @@ namespace vcpkg::Parse
 {
     struct TextRowCol
     {
-        TextRowCol() = default;
-        TextRowCol(int row, int column) : row(row), column(column) {}
+        constexpr TextRowCol() noexcept = default;
+        constexpr TextRowCol(int row, int column) noexcept : row(row), column(column) {}
         /// '0' indicates uninitialized; '1' is the first row.
         int row = 0;
         /// '0' indicates uninitialized; '1' is the first column.

--- a/toolsrc/include/vcpkg/textrowcol.h
+++ b/toolsrc/include/vcpkg/textrowcol.h
@@ -4,6 +4,8 @@ namespace vcpkg::Parse
 {
     struct TextRowCol
     {
+        TextRowCol() = default;
+        TextRowCol(int row, int column) : row(row), column(column) {}
         /// '0' indicates uninitialized; '1' is the first row.
         int row = 0;
         /// '0' indicates uninitialized; '1' is the first column.

